src/net_processing.cpp

```diff
@@ -2011,8 +2011,13 @@ void PeerManagerImpl::RelayTransaction(const uint256& txid, const uint256& wtxid
         auto tx_relay = peer.GetTxRelay();
         if (!tx_relay) continue;
 
-        const uint256& hash{peer.m_wtxid_relay ? wtxid : txid};
         LOCK(tx_relay->m_tx_inventory_mutex);
+        // Only queue transactions for announcement once the version handshake
+        // is completed. The time of arrival for these transactions is
+        // otherwise at risk of leaking to a spy.
```

Comment:
I can't fully grasp the exact risk here... How does it help the spy to learn something new, as opposed to listening/analyzing *after* the handshake?

Reply:
There is no delay, so they can poll all announcements in real time by just creating a new incoming connection in a loop

Reply:
Yeah, the comment from the following commit also helped :) Feel free to resolve this. This is a great finding.

I can only suggest explaining it here, and not in the following comment, because IMO the justification belongs where the transactions are added.

Reply:
Improved the comment in this location a little bit.